WatchApp Extension/Extensions/CLKComplicationTemplate.swift

```diff
@@ -10,37 +10,83 @@ import ClockKit
 import HealthKit
 import LoopKit
 import Foundation
-
+import LoopCore
 
 extension CLKComplicationTemplate {
 
-    static func templateForFamily(_ family: CLKComplicationFamily, from context: WatchContext, chartGenerator makeChart: () -> UIImage?) -> CLKComplicationTemplate? {
+    static func templateForFamily(
+        _ family: CLKComplicationFamily,
+        from context: WatchContext,
+        at date: Date,
+        recencyInterval: TimeInterval,
+        chartGenerator makeChart: () -> UIImage?
+    ) -> CLKComplicationTemplate? {
         guard let glucose = context.glucose, let unit = context.preferredGlucoseUnit else {
             return nil
         }
-
-        return templateForFamily(family, glucose: glucose, unit: unit, date: context.glucoseDate, trend: context.glucoseTrend, eventualGlucose: context.eventualGlucose, chartGenerator: makeChart)
+        
+        return templateForFamily(family,
+            glucose: glucose,
+            unit: unit,
+            glucoseDate: context.glucoseDate,
+            trend: context.glucoseTrend,
+            eventualGlucose: context.eventualGlucose,
+            at: date,
+            loopLastRunDate: context.loopLastRunDate,
+            recencyInterval: recencyInterval,
+            chartGenerator: makeChart)
     }
 
     static func templateForFamily(
         _ family: CLKComplicationFamily,
         glucose: HKQuantity,
         unit: HKUnit,
-        date: Date?,
+        glucoseDate: Date?,
         trend: GlucoseTrend?,
         eventualGlucose: HKQuantity?,
+        at date: Date,
+        loopLastRunDate: Date?,
+        recencyInterval: TimeInterval,
         chartGenerator makeChart: () -> UIImage?
     ) -> CLKComplicationTemplate? {
 
         let formatter = NumberFormatter.glucoseFormatter(for: unit)
-
-        guard let glucoseString = formatter.string(from: glucose.doubleValue(for: unit)),
-            let date = date else
-        {
+        
+        guard let glucoseDate = glucoseDate else {
             return nil
         }
+        
+        let glucoseString: String
+        let trendString: String
+        
+        let isGlucoseStale = date.timeIntervalSince(glucoseDate) > recencyInterval
+
+        if isGlucoseStale {
+            glucoseString = "---"
+            trendString = ""
+        } else {
+            guard let formattedGlucose = formatter.string(from: glucose.doubleValue(for: unit)) else {
+                return nil
+            }
+            glucoseString = formattedGlucose
+            trendString = trend?.symbol ?? " "
+        }
+        
+        let loopCompletionFreshness = LoopCompletionFreshness(lastCompletion: loopLastRunDate, at: date)
+        
+        let tintColor: UIColor
+        
+        switch loopCompletionFreshness {
+        case .fresh:
+            tintColor = .tintColor
+        case .aging:
+            tintColor = .agingColor
+        case .stale:
+            tintColor = .staleColor
```

Comment:
I'm curious why you don't use the `LoopCompletionFreshness` more often when determining the glucose freshness?

Also, isn't it possible that the glucose is considered `stale` using `recencyInterval`, but gets the style of `aging`? I think I'm missing the different between what the `recencyInterval` calls `stale` and what the `LoopCompletionFreshness` calls `stale`.

Reply:
Hopefully what I explained above explains this, but let me know if you still have questions.